1534-review-guide-for-replacing-servers (#1542)
Redocly/redocly-cli: docs/guides/replace-servers-url.md

## Prerequisites

- [Install @redocly/cli](../installation.md). We use version 1.12.0 in this tutorial.

- Use your favorite IDE for editing files (we use VS Code and have the [Redocly extension](../../redocly-openapi/index.md) installed).

> ## Create a demo specification file

Reviewer: ```suggestion
## Create a demo folder and definition file
```

> In this step, create an OpenAPI specification file.

Reviewer: ```suggestion
In this step, create an OpenAPI definition file.
```

Reviewer: Maybe we should say here why users are creating this folder and file. Something like, before we can create the plugin we need a sample definition document to test it with. In this step we will create the sample folder and file to use for testing our plugin later.

Reviewer 2: DESCRIPTION. Not definition. (sorry for shouting, caps lock was on but I figured it sort of made my point :laughing: )

1. Create a new directory and name it `replace-servers-demo`.

2. In `replace-servers-demo`, create an `original.yaml` file with the following content:

```yaml
openapi: 3.1.0

>     info:
>   version: 1.0.0
>   title: Custom decorators demo
>   description: The servers URL is replaced by the decorator during the `bundle` process.

Reviewer 3: ```suggestion
    info:
      version: 1.0.0
      title: Custom decorators demo
      description: The servers URL is replaced by the decorator during the `bundle` process.
```

servers:
  - url: 'https://example.com/api/v1'
paths:
  /status:
    get:
      summary: Get status
      operationId: getStatus
      security: []
      responses:
        '204':
          description: Status OK
        '400':
          description: Status not OK
```

3. Save the file.

## Create a custom plugin

In this step, create a custom plugin and define the decorator dependency.

> 1. Inside `replace-servers-demo`, create a directory called `plugins`.

Reviewer: ```suggestion
1. In the `replace-servers-demo` folder, create a directory called `plugins`.
```

};
```

3. Save the file.

{% admonition type="attention" %}
If you change the name of the decorators directory, make sure to also change it in line 1 of the [`plugin.js` file](#create-a-custom-plugin).
{% /admonition %}

## Configure the plugin for use

To use the decorator, register your your `plugins` and `decorators` in the [Redocly configuration file](../configuration/index.md).

> - In your Redocly configuration file, add this information:

Reviewer: Does the following yaml register the plugin and decorators?
